### PR TITLE
Integration tests for PrometheusScraper and impl of GetCPUUtilizationBreachDataPoints  

### PR DESCRIPTION
**Pull request by neerajb**
- Added integration tests using a locally downloaded prometheus server. Changes are in Makefile. Run `make test` to run the tests.
- Changed the unit tests to integration using Ginkgo. The test hosts a /metrics endpoint for hosting metrics via a registry. 
- Removed mockgen generated mocks as we are now using integration tests instead of mocking p8s api.
- Added implementation of GetCPUUtilizationBreachDataPoints which will be needed by breachMonitor.
- Fixed the promQL queries based on the tests.

### CODE REVIEW

File: metrics/scraper.go [hunk 1]
```diff
@@ -13,72 +13,186 @@ import (
...
+				hpaOwnerInfoMetric:    hpaOwnerInfoMetric,
+			}},
+		nil
+}
 
-	promAPI := v1.NewAPI(client)
+// GetAverageCPUUtilizationByWorkload returns the average CPU utilization for the given workload type and name in the
+// specified namespace, in the given time range.
+func (ps *PrometheusScraper) GetAverageCPUUtilizationByWorkload(namespace string,
+	workload string,
```
Comment (bharathguvvala) [inline metrics/scraper.go:80]: This is currently implemented for only workload type 'deployment'?

**Review by bharathguvvala — APPROVED** [review]
LGTM

File: metrics/scraper.go [hunk 2]
```diff
@@ -13,72 +13,186 @@ import (
...
 
 // PrometheusScraper is a Scraper implementation that scrapes metrics data from Prometheus.
 type PrometheusScraper struct {
-	api v1.API
+	api            v1.API
+	metricRegistry *MetricRegistry
+}
+
+type MetricRegistry struct {
+	utilizationMetric     string
```
[the remark this answers is not in the document]
Comment (neerajb) [reply metrics/scraper.go:37]: Yes, all the p8s interfacing should be done via this. We can add these functions as and when we need them.

File: metrics/scraper.go (hunk 1 above)
Comment (neerajb) [reply metrics/scraper.go:80]: namespace_workload_pod:kube_pod_owner:relabel relabels the workload type to deployment for both rollouts and deployments. This fn will work for both rollout and deployment

**Pull request merged**